### PR TITLE
[DEV APPROVED] 8664 - Use mas cms client gem for Article pages

### PR DESCRIPTION
**Pull request by tomas-stefano**
[TP](https://moneyadviceservice.tpondemand.com/entity/8664)

## Context
 
Frontend (our site) is still using the old and unnecessary implementation of communicating with our CMS.
This actually slows down our process of including new features and maintain old features because we need to change a lot of classes for a very simple thing.
 
## Solution
 
So this PR replaces the old implementation of getting **article pages (article pages only!)**  from CMS with the new mas cms client gem, which can be located here: github.com/moneyadviceservice/mas-cms-client/

## Caveats

* Since we change the client, some cassettes were updates to reflect the new user_agent mas-cms-client.
* This PR is waiting for this to be merged: https://github.com/moneyadviceservice/mas-cms-client/pull/9
When the above PR is merged, I will change the Gemfile.

---
This change is [Reviewable](https://reviewable.io/reviews/moneyadviceservice/frontend/1849)

### CODE REVIEW

**Comment by willhall88** [comment]
I'm getting 2 failing cucumber tests on this branch at the moment?

File: Gemfile [hunk 1]
```diff
@@ -42,7 +42,7 @@ gem 'sass-rails'
 gem 'statsd-ruby'
 gem 'postcode_anywhere-email_validation'
 gem 'turnout'
-gem 'mas-cms-client', '~> 1.1'
+gem 'mas-cms-client', git: 'git@github.com:moneyadviceservice/mas-cms-client.git', branch: '8664-add-article-preview-client' # '~> 1.1'
```
Comment (willhall88) [inline Gemfile]: don't forget to update this again once ready

**Comment by tomas-stefano** [comment]
@willhall88 everything updated and passing.

**Comment by willhall88** [comment]
👍  LGTM

**Comment by margOnline** [comment]
Can the commit history be cleaned up please.

**Comment by tomas-stefano** [comment]
@margOnline done!

File: app/controllers/debt_management_controller.rb [hunk 2]
```diff
@@ -7,9 +7,10 @@ class DebtManagementController < ApplicationController
...
     return @company_list if @company_list
-    @company_list = Core::ArticleReader.new(COMPANY_LIST_ARTICLE_ID).call do
-      not_found
-    end
+    @company_list = Mas::Cms::Article.find(
+      COMPANY_LIST_ARTICLE_ID,
+      locale: I18n.locale
+    )
     @company_list = ContentItemDecorator.decorate(@company_list)
   end
```
Comment (margOnline) [inline app/controllers/debt_management_controller.rb]: I think there is too much going on in this method. Perhaps, the decorating of the company list could be moved into a separate method, wdyt?

Comment (tomas-stefano) [reply app/controllers/debt_management_controller.rb]: Humm, seeing for the second time it makes sense move the cms find to a private method. Wdyt:

```
  @company_list = ContentItemDecorator.decorate(resource)

   private
   def resource
     Mas::Cms::Article.find(
       COMPANY_LIST_ARTICLE_ID,
       locale: I18n.locale
     )
   end
```

File: app/decorators/content_item_decorator.rb [hunk 3]
```diff
@@ -21,7 +21,8 @@ def footer_alternate_options
   end
 
   def canonical_url
-    h.send("#{object.class.to_s.underscore.gsub('core/', '')}_url", object.id)
+    resource_path = "#{object.class.to_s.demodulize.underscore.chomp('_preview')}_url"
```
Comment (margOnline) [inline app/decorators/content_item_decorator.rb]: 👍 for use of demodulize

File: app/controllers/debt_management_controller.rb (hunk 2 above)
Comment (margOnline) [reply app/controllers/debt_management_controller.rb]: Yes, that is more readable, but in the show method, right?

Comment (tomas-stefano) [reply app/controllers/debt_management_controller.rb]: ♻️

File: app/controllers/debt_management_controller.rb [hunk 4]
```diff
@@ -6,11 +6,7 @@ class DebtManagementController < ApplicationController
   private
 
   def company_list
-    return @company_list if @company_list
-    @company_list = Core::ArticleReader.new(COMPANY_LIST_ARTICLE_ID).call do
-      not_found
-    end
-    @company_list = ContentItemDecorator.decorate(@company_list)
+    @company_list ||= ContentItemDecorator.decorate(resource)
```
Comment (margOnline) [inline app/controllers/debt_management_controller.rb]: 😍

**Comment by margOnline** [comment]
lgtm 👍

**Pull request merged**